cpp/ql/test/library-tests/string_concat/strconcat.expected

```diff
@@ -0,0 +1,22 @@
+| concat.cpp:23:27:23:27 | call to operator+ | concat.cpp:23:22:23:25 | str1 | concat.cpp:23:22:23:31 | call to operator+ |
+| concat.cpp:23:27:23:27 | call to operator+ | concat.cpp:23:22:23:25 | str1 | concat.cpp:23:22:23:31 | call to operator+ |
+| concat.cpp:23:27:23:27 | call to operator+ | concat.cpp:23:22:23:25 | str1 | concat.cpp:23:27:23:27 | call to operator+ |
```

Comment:
Yes. I am outputting the operand and the result node. For the + operator, I'm guessing because of how the + is nested, it is picking several possibilities. I couldn't really understand why (if you run the test you will see exactly what I mean). I couldn't think of how to fix it,, so I left it for now and its not really wrong.

Reply:
Hmm, I'll take another look...

Reply:
Ah, it's because the mapping between `Expr`s and `DataFlow::Node`s isn't 1:1.  In this case there are three `DataFlow::Node`s corresponding to one `Call` for the result node.  It's by design, it probably could be cleaner, but I don't think it's your fault and I don't think it will cause any problems.

Reply:
I had considered making the 'result' of the concat just an expr return, but this is tricky potentially given an input parameter can also be an output parameter. I wanted to make a distinction so I decided dataflow node was best here, but for getting the operand I decided expressions were fine.

Reply:
> Ah, it's because the mapping between `Expr`s and `DataFlow::Node`s isn't 1:1. In this case there are three `DataFlow::Node`s corresponding to one `Call` for the result node. It's by design, it probably could be cleaner, but I don't think it's your fault and I don't think it will cause any problems.

That doesn't sound quite right. There should only be one dataflow node for a given `Call`. That is, there should only be one Node `n` such that `n.asExpr()` gives back that `Call`. Of course, if `Call` returns a pointer there can also be multiple `(n, i)` pairs such that `n.asIndirectExpr(i)` gives back the `i`'th indirection of `Call`.

Reply:
I've investigated this a bit, and I think there's a bug in our expression <-> dataflow node mapping in this case. I'm still trying to understand why this happens, but just to be clear: It's _not_ by design that there should be multiple dataflow nodes for the same expression. We've fought very hard to get rid of such cases because they lead to duplicated query results such as this one.

I don't think this should block anything on this PR, though.

Reply:
Thanks for the correction.